langkit/templates/astnode_types_ada.mako

```diff
@@ -290,9 +292,23 @@
    ## Lexical Environments Handling ##
    ###################################
 
-   % if cls.env_spec:
+   % if cls.env_spec and cls.env_spec.actions:
 
-   <% call_prop = cls.env_spec._render_field_access %>
+   <%
+      call_prop = cls.env_spec._render_field_access
+
+      # Whether the initial env for this node is dynamic
+      has_dyn_env = cls.env_spec.initial_env
```

Comment:
`has_dyn_env = cls.env_spec.initial_env is not None` makes it clearer IMO

Reply:
Ok, I’ll update it